drivers/hmis/app/models/hmis/ce/match/cde_field_map.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# resolve custom data elements (CDEs)
+module Hmis::Ce::Match
+  class CdeFieldMap
+    def instance_value(client, field)
+      cded = parse_entity_type(field)
+
+      # choose the assessment that was most recently updated
+      record = client.custom_assessments.joins(:definition).
+        where(definition: { identifier: cded.form_definition_identifier }).
+        order(:date_updated, :id).
+        last
+      return nil unless record
+
+      # Get all CDE values for this definition
+      cdes = record.custom_data_elements.where(data_element_definition: cded)
+      return nil if cdes.empty?
+
+      if cded.repeats
+        # For multi-valued CDEs, return an array of values
+        cdes.map { |cde| cded.read_value_from(cde) }.compact
+      else
+        # For single-valued CDEs, return the single value
+        cded.read_value_from(cdes.first)
+      end
+    end
+
+    # arel not supported.
+    # Enhancement: if expression requires certain CDEs to be present then we could take a rough pass to filter out clients that lack the relevant assessments
+    def arel_field(_field)
+      nil
+    end
+
+    protected
+
+    # custom_assessment.xyz
+    def parse_entity_type(field)
+      entity_type, cde_key = field.split('.', 2)
+
+      klass = case entity_type
+      when 'custom_assessment'
+        Hmis::Hud::CustomAssessment
+      # TBD: add support for CDEs on other entities such as Enrollments and Client
+      else
+        raise ArgumentError, "Unknown entity in field \"#{field}\""
+      end
+
+      cded = cded_lookup.dig(klass.sti_name, cde_key)
+      raise ArgumentError, "Unknown CDE in field \"#{field}\"" unless cded
+
+      cded
+    end
+
+    def cded_lookup
+      # perhaps this should be constrained by data source
```

Comment:
Hmm. As written, perhaps yes, since we're still in the world of dealing with source clients. 

However with https://github.com/open-path/Green-River/issues/7671 coming in, this will need to be slightly changed since `client` will be a destination client. The above `client.custom_assessments...` query will need to be adjusted to fetch custom assessments for all associated source clients, which may span data sources.

That's all to say, I _think_ we should leave this lookup as-is, including all data sources, in anticipation of issue 7671. For better or worse, the uniqueness constraint on `owner_type,key` spans across data sources.

cc @martha since she'll likely be working on that ticket.

Reply:
happy to discuss more, but I think leaving as-is for now makes sense

Reply:
notes from discussion: I'm going to adjust the resolver to support destination clients in anticipation of 7671